tenant/src/main/scala/com/improving/app/tenant/domain/Tenant.scala

```diff
@@ -57,97 +61,125 @@ object Tenant {
     )
   }
 
-  private val commandHandler: (TenantState, TenantCommand) => ReplyEffect[TenantEnvelope, TenantState] = {
-    (state, command) =>
-      val result: Either[Error, TenantEnvelope] = state match {
-        case UninitializedTenant =>
-          command.request match {
-            case x: EstablishTenant  => establishTenant(x)
-            case _: GetOrganizations => getOrganizations()
-            case _                   => Left(StateError("Tenant is not established"))
-          }
-        case draftState: DraftTenant =>
-          command.request match {
-            case _: EstablishTenant  => Left(StateError("Tenant is already established"))
-            case x: ActivateTenant   => activateTenant(draftState, x)
-            case x: SuspendTenant    => suspendTenant(draftState, x)
-            case x: EditInfo         => editInfo(draftState, x)
-            case _: GetOrganizations => getOrganizations(Some(draftState))
-            case x: TerminateTenant  => terminateTenant(draftState, x)
-            case _                   => Left(StateError("Command is not supported"))
-          }
-        case establishedState: EstablishedTenant =>
-          establishedState match {
-            case activeTenantState: ActiveTenant =>
-              command.request match {
-                case _: EstablishTenant  => Left(StateError("Tenant is already established"))
-                case _: ActivateTenant   => Left(StateError("Active tenants may not transition to the Active state"))
-                case x: SuspendTenant    => suspendTenant(establishedState, x)
-                case x: EditInfo         => editInfo(establishedState, x)
-                case _: GetOrganizations => getOrganizations(Some(activeTenantState))
-                case x: TerminateTenant  => terminateTenant(establishedState, x)
-                case _                   => Left(StateError("Command is not supported"))
+  private val commandHandler: (TenantState, TenantRequestEnvelope) => ReplyEffect[TenantResponse, TenantState] = {
+    (state, envelope) =>
+      val requestErrors: Either[ReplyEffect[TenantResponse, TenantState], TenantRequestPB with TenantRequest] =
+        envelope.request match {
+          case r: TenantRequest => Right(r)
+          case _                => Left(Effect.reply(envelope.replyTo)(StatusReply.Error("Message was not a TenantRequest")))
+        }
+
+      requestErrors.map(tenantRequestValidator) match {
+        case Right(None) =>
+          def getCommandErrors(
+              errorsOrResponse: Either[Error, TenantResponse]
+          ): Either[ReplyEffect[TenantResponse, TenantState], Either[Error, TenantResponse]] =
+            envelope.request match {
+              case r: TenantCommand =>
+                tenantCommandValidator(r) match {
+                  case None => Right(errorsOrResponse)
+                  case Some(err: ValidationError) =>
```

Comment:
This should just be a `Left[ValidationError]` and then that transformed into the `ReplyEffect` in the same place all the other errors are handled